lib/modules/dosomething/dosomething_reportback/dosomething_reportback.cron.inc

```diff
@@ -10,13 +10,15 @@
  */
 function dosomething_reportback_cron() {
   // Get all reportbacks updated in the past hour.
-  $results = db_query("SELECT n.nid, SUM(rb.quantity) as total
+  $results = db_query("SELECT n.nid, SUM(rb.quantity) as total, run.field_current_run_target_id as run_nid
                        FROM node n
-                       INNER JOIN field_data_field_campaign_status s on n.nid = s.entity_id
+                       INNER JOIN field_data_field_campaign_status s on n.nid = s.entity_id AND n.language = s.language
                        INNER JOIN dosomething_reportback rb on n.nid = rb.nid
+                       INNER JOIN field_data_field_current_run run on n.nid = run.entity_id
```

Comment:
since this field is translatable, you may have to join on both id and language here as well